pkg/kds/global/components.go

```diff
@@ -80,22 +103,44 @@ func Setup(rt runtime.Runtime) error {
 		}()
 		return nil
 	})
+
+	onGlobalToZoneSyncConnect := global_service.OnGlobalToZoneSyncConnectFunc(func(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errChan chan error) {
+		clientId, err := util.ClientIDFromIncomingCtx(stream.Context())
+		if err != nil {
+			errChan <- err
+		}
+		log := kdsDeltaGlobalLog.WithValues("peer-id", clientId)
+		if err := createZoneIfAbsent(log, clientId, rt.ResourceManager()); err != nil {
+			log.Error(err, "Global CP could not create a zone")
+			errChan <- err
+		}
+		if err := kdsServerV2.GlobalToZoneSync(stream); err != nil {
```

Comment:
This doesn't feel right, this callback is already called from the `GlobalToZoneSync` method, so it looks like you're calling it again but from a different goroutine. 

I'm not sure we need `onGlobalToZoneSyncConnect` callback at all. It's required for old KDS implementation because of the multiplex, but here we probably don't need it.

Reply:
It's required to provide stream to delta xDS server. `kdsServerV2` is only available here and without this call session won't be intercepted by go control-plane server